code/modules/clothing/under/accessories/badges.dm

```diff
@@ -0,0 +1,205 @@
+	. = ..()
+	if(!.)
+		return
+
+	if(!isnull(attach_to.atom_storage))
+		if(user)
+			attach_to.balloon_alert(user, "not compatible!")
+		return FALSE
+	return TRUE
+
+/obj/item/clothing/accessory/pocketprotector/full
+
+/obj/item/clothing/accessory/pocketprotector/full/Initialize(mapload)
+	. = ..()
+	new /obj/item/pen/red(src)
+	new /obj/item/pen(src)
+	new /obj/item/pen/blue(src)
+
+/obj/item/clothing/accessory/pocketprotector/cosmetology
+
+/obj/item/clothing/accessory/pocketprotector/cosmetology/Initialize(mapload)
+	. = ..()
+	for(var/i in 1 to 3)
+		new /obj/item/lipstick/random(src)
+
+/obj/item/clothing/accessory/dogtag
+	name = "Dogtag"
+	desc = "Can't wear a collar, but this is fine?"
+	icon_state = "allergy"
+	attachment_slot = NONE // actually NECK but that doesn't make sense
+	/// What message is displayed when our dogtags / its clothes / its wearer is examined
+	var/display = "Nothing!"
+
+/obj/item/clothing/accessory/dogtag/examine(mob/user)
+	. = ..()
+	. += display
+
+// Examining the clothes will display the examine message of the dogtag
+/obj/item/clothing/accessory/dogtag/attach(obj/item/clothing/under/attach_to, mob/living/attacher)
+	. = ..()
+	if(!.)
+		return
+	RegisterSignal(attach_to, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+
+/obj/item/clothing/accessory/dogtag/detach(obj/item/clothing/under/detach_from)
+	. = ..()
+	UnregisterSignal(detach_from, COMSIG_ATOM_EXAMINE)
+
+// Double examining the person wearing the clothes will display the examine message of the dogtag
+/obj/item/clothing/accessory/dogtag/accessory_equipped(obj/item/clothing/under/clothes, mob/living/user)
+	RegisterSignal(user, COMSIG_ATOM_EXAMINE_MORE, PROC_REF(on_examine))
+
+/obj/item/clothing/accessory/dogtag/accessory_dropped(obj/item/clothing/under/clothes, mob/living/user)
+	UnregisterSignal(user, COMSIG_ATOM_EXAMINE_MORE)
+
+/// Adds the examine message to the clothes and mob.
+/obj/item/clothing/accessory/dogtag/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	// Only show the examine message if we're close (2 tiles)
+	if(!IN_GIVEN_RANGE(get_turf(user), get_turf(src), 2))
+		return
+
+	if(ismob(source))
+		// Examining a mob wearing the clothes, wearing the dogtag will also show the message
+		examine_list += "A dogtag is hanging around [source.p_their()] neck: [display]"
+	else
+		examine_list += "A dogtag is attached to [source]: [display]"
+
+/obj/item/clothing/accessory/dogtag/allergy
+	name = "Allergy dogtag"
+	desc = "A dogtag with a listing of allergies."
+
+/obj/item/clothing/accessory/dogtag/allergy/Initialize(mapload, allergy_string)
+	. = ..()
+	if(allergy_string)
+		display = span_notice("The dogtag has a listing of allergies: [allergy_string]")
+	else
+		display = span_notice("The dogtag is all scratched up.")
+
+/// Reskins for the pride pin accessory, mapped by display name to icon state
```

Comment:
The inclusion of the generic Pride Pin accessory seems to be the cause of test failures as we do not have the DMIs implemented as seems to be the case.